phac-nml/irida-next
# Fix: Action links incorrectly enabled on empty table (DFCT0010080)

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR fixes the Action Links on Project and Group samples from being incorrectly enabled when No Samples associated with a Group or a Project.

This PR also refactors the Group and Project Samples tables.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/492127/e4891ab8-018f-4191-b2d7-37273a4bc06f)

![image](https://github.com/phac-nml/irida-next/assets/492127/cfd20787-24f7-4edc-99d9-bcdfe7f6a245)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Create an new Project without any Samples
2. Navigate to the Samples page for the new Project
3. Ensure that the action links are disabled and that an empty state message is shown
4. Create a new Group
5. Add the Project created above to the group
6. Navigate to the Samples page for the new Group
7. Ensure that the action links are disabled and that an empty state message is shown

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
